source/geometries/NextDemoSiPMBoard.cc

```diff
@@ -259,32 +293,40 @@ void NextDemoSiPMBoard::Construct()
   G4LogicalVolume* coating_logic;
 
   if (coating_thickn_ > 0.) {
-    G4String coating_name = "BOARD_COATING";
 
     G4double coating_posz = board_size_z/2. - coating_thickn_/2.;
 
     G4Box* coating_solid = new G4Box(coating_name, board_size_x/2.,
                                      board_size_y/2., coating_thickn_/2.);
 
-    G4Material* tpb = MaterialsList::TPB();
-    tpb->SetMaterialPropertiesTable(OpticalMaterialProperties::TPB());
-
     coating_logic = new G4LogicalVolume(coating_solid, tpb, coating_name);
 
     G4PVPlacement* coating_phys =
       new G4PVPlacement(nullptr, G4ThreeVector(0.,0.,coating_posz), coating_logic,
                         coating_name, board_logic, false, 0, true);
 
-    // Optical surface
-    G4OpticalSurface* coating_opsurf =
-      new G4OpticalSurface(coating_name + "_OPSURF", glisur, ground,
-                           dielectric_dielectric, .01);
-
     new G4LogicalBorderSurface("TEFLON_WLS_GAS_OPSURF", coating_phys,
                                mother_phys_, coating_opsurf);
     new G4LogicalBorderSurface("GAS_TEFLON_WLS_OPSURF", mother_phys_,
                                coating_phys, coating_opsurf);
 
+    // Coating holes
+    if (membrane_thickn_ == 0.){
+      if (hole_type_ == "rectangular"){
+
+        G4Box* board_coating_hole_solid = new G4Box("BOARD_COATING_HOLE", hole_x_/2., hole_y_/2., coating_thickn_/2.);
```

Comment:
I found a little bit missleading the name **BOARD_COATING_HOLE**. I would add some lines specifying the purpouse of this section.